galaxy/api/v2/views/collection.py

```diff
@@ -102,6 +107,7 @@ def post(self, request, *args, **kwargs):
         # TODO(cutwater): Merge Artifact and UploadCollectionSerializers
         namespace = self._get_namespace(data)
         self._check_namespace_access(namespace, request.user)
+        self._check_max_file_size(request.data['file'].size)
```

Comment:
The ultimate problem with this solution is that it checks file size after the file is uploaded to the server. It's probably acceptable as "better than nothing" solution, but please add comment \ note \ warning.